### DISCUSSION TITLE
Styling applied only once afer page reception?

### DISCUSSION THREAD

**Discussion by R-W-C** [ankit/stylebot#791] [Q&A]
Hello,

I have got the impression  the styling is applied only once after page reception. Is that correct? Or did I miss something?

If yes, wouldn't it be useful (and in my opinion) logical to have a button available to reapply the styling? 
A lot of times the page is changed because some javascript is being executed (AJAX calls). Think of adding or modifying divs on the page or loading new content for iframes. And maybe it could be a good idea to have the styling reapplied with a configurable interval. Thoughts?

**Comment by generrosity** [comment]
I'm pretty sure that the CSS is available immediately 🤔 It definitely works like a normal CSS file 'dynamically' as I've been using it with various facebook and Jira that async ajax, or editing tags manually.

I've found if you have sub-frames that you need to refresh the sub-frame if you have changed something.

**Comment by m-dwork** [comment]
Sometimes  the page will change the URL, without actually reloading the page, which is common with many modern web apps. So for the style to apply, you need to reload the page, or turn the style off and then on again in stylebot, so that stylebot can apply the style that matches the url.

Stylebot still dynamically updates pages, as you can see when you update its css. The issue is with the url mismatch, which will have stylebot displaying and applying the style for the previous or original url.
